packages/babel-parser/src/parser/expression.js

```diff
@@ -1166,10 +1178,21 @@ export default class ExpressionParser extends LValParser {
       val = exprList[0];
     }
 
-    this.addExtra(val, "parenthesized", true);
-    this.addExtra(val, "parenStart", startPos);
+    if (!this.options.createParenthesizedExpressions) {
+      this.addExtra(val, "parenthesized", true);
+      this.addExtra(val, "parenStart", startPos);
+      return val;
+    }
 
-    return val;
+    const parenExpression = this.startNodeAt(startPos, startLoc);
+    parenExpression.expression = val;
+    this.finishNodeAt(
```

Comment:
Isn't using `this.finishNode` enough? The end position should be the correct one at this point I think.